src/frequenz/repo/config/mkdocs.py

```diff
@@ -0,0 +1,76 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Generate the code reference pages.
+
+It uses the following `mkdocs` plugins:
+
+* `mkdocs-gen-files` to generate the API documentation pages.
+* `mkdocs-literate-nav` to make use of the generate `SUMMARY.md` file.
+
+Based on the recipe at:
+https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages
+"""
+
+from pathlib import Path
+from typing import Tuple
+
+import mkdocs_gen_files
+
+
+def _is_internal(path_parts: Tuple[str, ...]) -> bool:
+    """Tell if the path is internal judging by the parts.
+
+    Args:
+        path_parts: Path.parts of the path to check.
+
+    Returns:
+        True if the path is internal.
+    """
+
+    def with_underscore_not_init(part: str) -> bool:
+        return part.startswith("_") and part != "__init__"
+
+    return any(p for p in path_parts if with_underscore_not_init(p))
+
+
+def generate_api_pages(src_path: str = "src", dst_path: str = "reference") -> None:
+    """Generate API documentation pages for the code.
+
+    Internal modules (those starting with an underscore except from `__init__`) are
+    not included.
+
+    A summary page is generated as `SUMMARY.md` which is compatible with the
```

Comment:
What about things like `__iter__` and other ... dunder methods?

Reply:
This only affects files, so there is only `__init__.py` as a special case. It's dealing with the directory structure of modules and packages.

Reply:
what about `__main__` in that case?

Reply:
Mmm, I never seen `__main__.py` but it does seems to exist: https://docs.python.org/3/library/__main__.html#main-py-in-python-packages

But still I'm not sure if it is a good candidate to generate docs for, docs are mainly generated as a API reference, and some code that is supposed to be the CLI interface for a package is probably not designed to be used as a library, even when other modules can potentially `import __main__`. Seems all pretty obscure IMHO.

Reply:
I am using it in our actor for example https://github.com/frequenz-io/frequenz-actor-fcr-balancing/blob/main/src/frequenz/app/fcr_balancing/__main__.py

Reply:
But the actor in the long run shouldn't have a `__main__.py`, as it is not an app, is a library. And even if it were long term, do you want the main module to appear in the generated API docs? What for?

Reply:
I don't have a good response, I just noticed it :)